RInAppMessaging/Classes/Router.swift

```diff
@@ -29,21 +28,17 @@ internal class Router: RouterType {
         self.dependencyManager = dependencyManager
     }
 
-    func discardDisplayedCampaign() -> Campaign? {
-        return displayQueue.sync {
-            let discard: () -> Campaign? = {
-                guard let rootView = UIApplication.shared.getKeyWindow() else {
-                    return nil
+    func discardDisplayedCampaign() {
+        displayQueue.sync {
+            DispatchQueue.main.async {
```

Comment:
Is the main q needed because you're calling `UIApplication` API?

Reply:
Yes. especially the `removeFromSuperview()` part